llvm/lib/SYCLLowerIR/ESIMD/LowerESIMDSlmReservation.cpp

```diff
@@ -635,6 +635,11 @@ size_t lowerSLMReservationCalls(Module &M) {
   // - now set each kernel's SLMSize metadata to the pre-calculated value
   for (auto &E : Kernel2MaxSLM) {
     int MaxSLM = E.second;
+    // Clamp negative values to 0. MaxSLM could have been not estimated, e.g.
```

Comment:
do we know why it worked previously without this change?

Reply:
> do we know why it worked previously without this change?

AFAIU is wasn't working and there is an internal report for 'not enough SLM', not 100% though if this patch relates with this report.

Reply:
if its the same report im thinking of, i thouguht it did work previously before https://github.com/intel/llvm/commit/e7674019af00f50615e3e8a48276390334cfbbc4, but its not a big deal i was just wondering

Reply:
If MaxSLM is possibly not estimated, is the use of MaxSLM on line 587 also invalid?  Should clamping be done earlier at line 578?

Reply:
> do we know why it worked previously without this change?

Yes, before https://github.com/intel/llvm/commit/e7674019af00f50615e3e8a48276390334cfbbc4, `VCSLMSize` was set to `0`.

Reply:
> If MaxSLM is possibly not estimated, is the use of MaxSLM on line 587 also invalid? Should clamping be done earlier at line 578?

No, missing estimation happens only with `slm_init` where it is allowed to have non-constant size (spec constant). Offset is `0` for slm_init, so it is not a problem.
`-1` is used as an Identity value throughout the computation, I was hesitant to change that.